packages/core/src/browser/view-container.ts

```diff
@@ -531,6 +631,38 @@ export class ViewContainer extends BaseWidget implements StatefulWidget, Applica
         }
     }
 
+    appendPart(part: ViewContainerPart, index: number): void {
+        const existing = this.toRemoveWidgets.get(part.wrapped.id);
+        if (existing) {
+            return;
+        }
+        const newPart = this.createNewPartFromExisting(part);
+        this.attachNewPart(newPart, index);
+        part.fireOnDidRecreate(newPart, this);
+    }
+
+    protected moveBetweenContainers(fromIndex: number, toIndex: number, fromContainer: ViewContainer): void {
+        const partToMove = fromContainer.getParts()[fromIndex];
+        fromContainer.removeWidget(partToMove);
```

Comment:
I am currently in the process of finding out what causes the weird layouting behavior after reloading.

One thing I noticed during debugging is that this `fromContainer.removeWidget(partToMove);` call never does anything. This is due to [this line](https://github.com/eclipse-theia/theia/blob/b6b0f1be2375fd454fefbbbf28895bdffe06fc69/packages/core/src/browser/view-container.ts#L372) which uses the id of the wrapped element. But here we call remove on the wrapper instead, which is not part of the `toRemoveWidgets` map.

I'll continue investigating what's wrong here.

Reply:
@msujew thanks for helping, I really missed that `removeWidget` was never called, I have fixed it but the issue still exist I'm trying my best..., Have you discovered some more hints?

Reply:
Yeah, I spent an hour trying to fix it but to no avail as well. 

Looking at the html source, I noticed that the last `ViewContainerPart` isn't rendered correctly. Somehow, phosphor does not correctly render its top and height style properties. That's why it shows up at the top and basically shadows every other element in the `ViewContainer`. I couldn't figure out why this happens though. I'll take another look on monday.

Reply:
@msujew Thank you, I have also noticed that it has no `top` style and its handle is hidden (might not right in all cases)
Actually if there will be no solution we are able to completely disable drag & drop for the `debugger` container  (which is the only one who makes these troubles) as done for `Extensions`, Don't we?
cc @colin-grant-work